hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -0,0 +1,291 @@
+	}
+	dest := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      src.Name,
+		},
+	}
+	_, err := createOrUpdate(ctx, c, dest, func() error {
+		srcData, srcHasData := src.Data["ibm-credentials.env"]
+		if !srcHasData {
+			return fmt.Errorf("hostedcluster cloud controller provider credentials secret %q must have a credentials key", src.Name)
+		}
+		dest.Type = corev1.SecretTypeOpaque
+		if dest.Data == nil {
+			dest.Data = map[string][]byte{}
+		}
+		dest.Data["ibm-credentials.env"] = srcData
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile cloud controller provider creds: %w", err)
+	}
+
+	// Reconcile the platform provider node pool management credentials secret by
+	// resolving  the reference from the HostedCluster and syncing the secret in
+	// the control plane namespace.
+	err = c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.PowerVS.NodePoolManagementCreds.Name}, &src)
+	if err != nil {
+		return fmt.Errorf("failed to get node pool provider creds %s: %w", hcluster.Spec.Platform.PowerVS.NodePoolManagementCreds.Name, err)
+	}
+	dest = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      src.Name,
+		},
+	}
+	_, err = createOrUpdate(ctx, c, dest, func() error {
+		srcData, srcHasData := src.Data["ibm-credentials.env"]
+		if !srcHasData {
+			return fmt.Errorf("node pool provider credentials secret %q is missing credentials key", src.Name)
+		}
+		dest.Type = corev1.SecretTypeOpaque
+		if dest.Data == nil {
+			dest.Data = map[string][]byte{}
+		}
+		dest.Data["ibm-credentials.env"] = srcData
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile node pool provider creds: %w", err)
+	}
+
+	// Reconcile the platform provider node pool management credentials secret by
+	// resolving  the reference from the HostedCluster and syncing the secret in
+	// the control plane namespace.
+	err = c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.PowerVS.ControlPlaneOperatorCreds.Name}, &src)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane operator provider creds %s: %w", hcluster.Spec.Platform.PowerVS.ControlPlaneOperatorCreds.Name, err)
+	}
+	dest = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      src.Name,
+		},
+	}
+	_, err = createOrUpdate(ctx, c, dest, func() error {
+		srcData, srcHasData := src.Data["ibm-credentials.env"]
+		if !srcHasData {
+			return fmt.Errorf("control plane operator provider credentials secret %q is missing credentials key", src.Name)
+		}
+		dest.Type = corev1.SecretTypeOpaque
+		if dest.Data == nil {
+			dest.Data = map[string][]byte{}
+		}
+		dest.Data["ibm-credentials.env"] = srcData
+		return nil
```

Comment:
where's the `ibm-credentials.env` path coming from? Is there any IBM docs we can ref?

Reply:
This is coming from https://github.com/IBM/go-sdk-core/blob/ca4450e2fb66d8a4b8bf72d0712407b9cf65a2bf/v5/core/config_utils.go#L35 which is used in most of the go sdk clients to load the authentication.

Reply:
It'd be nice to add a comment with a link

Reply:
added this information, ptal at latest amended commit.
```golang
        // This secret hosts the IBM Cloud credential with filename ibm-credentials.env
 	// used by the different components in the cluster like CAPI controller.
 	// The filename will be searched for within the program's working directory, and then the OS's
 	// current user directory. This filename can be overridden with the help of IBM_CREDENTIALS_FILE environment variable
```